nixos/modules/programs/xppen.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib; let
+  cfg = config.programs.xppen;
+in
+
+{
+  options.programs.xppen = {
+    enable = mkEnableOption (lib.mdDoc "XPPen PenTablet application");
+    package = mkPackageOption pkgs "xppen" { };
+  };
+
+  config = mkIf cfg.enable {
+    hardware.uinput.enable = true;
+
+    environment.systemPackages = [
+      cfg.package
+    ];
+
+    services.udev.packages = [ cfg.package ];
+
+    system.activationScripts.xppen.text = with pkgs; ''
```

Comment:
Does this need to be a `activationScripts`?
I think, a systemd oneshot service, that would only need to be restarted if `xppen` changed would be better, than a activationScripts that needs to always run on activation.